diff -Naur partitionmanager-2.1.0/src/partitionmanager.kcfg partitionmanager-2.1.0.tpg/src/partitionmanager.kcfg
--- partitionmanager-2.1.0/src/partitionmanager.kcfg	2016-03-11 00:11:40.000000000 +0000
+++ partitionmanager-2.1.0.tpg/src/partitionmanager.kcfg	2016-05-09 17:41:14.885985319 +0000
@@ -98,7 +98,7 @@
 			<default>pmlibpartedbackendplugin</default>
 		</entry>
 		<entry key="defaultFileSystem" type="Int">
-			<default>FileSystem::Ext3</default>
+			<default>FileSystem::Ext4</default>
 		</entry>
 		<entry key="preferredUnit" type="Int">
 			<label context="@label">Preferred unit:</label>
