--- partitionmanager-2.2.0/src/gui/mainwindow.cpp.orig	2016-06-06 21:03:42.294047754 +0100
+++ partitionmanager-2.2.0/src/gui/mainwindow.cpp	2016-06-06 23:08:25.306524009 +0100
@@ -1100,6 +1100,13 @@
 
     KLocalizedString rval;
 
+    KLocalizedString templ = kxi18n("<tr>"
+                          "<td>%1</td>"
+                          "<td>%2</td>"
+                          "<td>%3</td>"
+                          "<td><a href=\"%4\">%4</a></td>"
+                          "</tr>");
+
     foreach(const PartitionNode * node, parent->children()) {
         const Partition* p = dynamic_cast<const Partition*>(node);
 
@@ -1107,19 +1114,39 @@
             continue;
 
         if (node->children().size() > 0)
-            rval = kxi18n("%1%2").subs(rval).subs(checkSupportInNode(node));
+        {
+            KLocalizedString childRval = checkSupportInNode(node);
+            if(!childRval.isEmpty())
+            {
+              if(rval.isEmpty())
+              {
+                rval = childRval;
+              }
+              else
+              {
+                rval = kxi18n("%1%2").subs(rval).subs(childRval);
+              }
+           }
+        }
 
         if (!p->fileSystem().supportToolFound() && !p->fileSystem().supportToolName().name.isEmpty())
-            rval = kxi18n("%1%2").subs(rval).subs(kxi18n("<tr>"
-                                   "<td>%1</td>"
-                                   "<td>%2</td>"
-                                   "<td>%3</td>"
-                                   "<td><a href=\"%4\">%4</a></td>"
-                                   "</tr>")
+        {
+            if(!rval.isEmpty())
+            {
+              rval = kxi18n("%1%2").subs(rval).subs((templ
                  .subs(p->deviceNode())
                  .subs(p->fileSystem().name())
                  .subs(p->fileSystem().supportToolName().name)
-                 .subs(p->fileSystem().supportToolName().url.toString()));
+                 .subs(p->fileSystem().supportToolName().url.toString()) ));
+            }
+            else
+            {
+              rval = templ.subs(p->deviceNode())
+                 .subs(p->fileSystem().name())
+                 .subs(p->fileSystem().supportToolName().name)
+                 .subs(p->fileSystem().supportToolName().url.toString());
+            }
+        }
     }
 
     return rval;
@@ -1132,9 +1159,16 @@
 
     foreach(const Device * d, operationStack().previewDevices()) {
         supportInNode = checkSupportInNode(d->partitionTable());
-        if (supportInNode.toString() != QString()) {
+        if (!supportInNode.isEmpty()) {
             missingSupportTools = true;
-            supportList = kxi18n("%1%2").subs(supportList).subs(supportInNode);
+            if(!supportList.isEmpty())
+            {
+              supportList = kxi18n("%1%2").subs(supportList).subs(supportInNode);
+            }
+            else
+            {
+              supportList = supportInNode;
+            }
         }
     }
 
